examples/bert/run_glue_finetuning.py

```diff
@@ -0,0 +1,242 @@
+
+flags.DEFINE_float("learning_rate", 2e-5, "The initial learning rate for Adam.")
+
+flags.DEFINE_integer("max_seq_length", 128, "Maximum sequence length.")
+
+
+def pack_inputs(
+    inputs,
+    seq_length,
+    start_of_sequence_id,
+    end_of_segment_id,
+    padding_id,
+):
+    # In case inputs weren"t truncated (as they should have been),
+    # fall back to some ad-hoc truncation.
+    trimmed_segments = tftext.RoundRobinTrimmer(
+        seq_length - len(inputs) - 1
+    ).trim(inputs)
+    # Combine segments.
+    segments_combined, segment_ids = tftext.combine_segments(
+        trimmed_segments,
+        start_of_sequence_id=start_of_sequence_id,
+        end_of_segment_id=end_of_segment_id,
+    )
+    # Pad to dense Tensors.
+    input_word_ids, _ = tftext.pad_model_inputs(
+        segments_combined, seq_length, pad_value=padding_id
+    )
+    input_type_ids, input_mask = tftext.pad_model_inputs(
+        segment_ids, seq_length, pad_value=0
+    )
+    # Assemble nest of input tensors as expected by BERT model.
+    return dict(
+        input_ids=input_word_ids,
+        input_mask=input_mask,
+        segment_ids=input_type_ids,
+    )
+
+
+def load_data(task_name):
+    if task_name in ("cola", "sst2"):
+        feature_names = ("sentence",)
+    elif task_name in ("mrpc", "stsb", "rte", "wnli"):
+        feature_names = ("sentence1", "sentence2")
+    elif task_name in ("mnli", "mnli_matched", "mnli_mismatched"):
+        feature_names = ("premise", "hypothesis")
+    elif task_name in "qnli":
+        feature_names = ("question", "sentence")
+    elif task_name in "qqp":
+        feature_names = ("question1", "question2")
+    else:
+        raise ValueError(f"Unkown task_name {task_name}.")
+
+    test_suffix = ""
+    if task_name in ("mnli", "mnli_matched"):
+        # For "mnli", just run default to "mnli_matched".
+        task_name = "mnli"
+        test_suffix = "_matched"
+    elif task_name in ("mnli_mismatched",):
+        task_name = "mnli"
+        test_suffix = "_mismatched"
+
+    def to_tf_dataset(split):
+        # Format each sample as a tuple of string features and an int label.
+        features = tuple([split[f] for f in feature_names])
+        label = tf.cast(split["label"], tf.int32)
+        return tf.data.Dataset.from_tensor_slices((features, label))
+
+    data = datasets.load_dataset("glue", task_name)
+    data.set_format(type="tensorflow")
+    train_ds = to_tf_dataset(data["train"])
+    test_ds = to_tf_dataset(data["test" + test_suffix])
+    validation_ds = to_tf_dataset(data["validation" + test_suffix])
+    return train_ds, test_ds, validation_ds
+
+
+class BertClassificationFinetuner(keras.Model):
+    """Adds a classification head to a pre-trained BERT model for finetuning"""
+
+    def __init__(self, bert_model, hidden_size, num_classes, **kwargs):
+        super().__init__(**kwargs)
+        self._bert_model = bert_model
```

Comment:
Nit: no underscore should be preferred (attributes that the match names of the keyword arguments)

Reply:
done